### PR TITLE
Add API for Wasm modules heap footprint

### PR DESCRIPTION
**Pull request by mikevoronov**
This PR introduces a new API that intended to show Wasm modules heap usage statistic:
```rust
pub struct HeapStatisticRecord<'module_name> {
    pub module_name: &'module_name str,
    pub memory_size: usize,
}

pub struct HeapStatistic<'module_name>(pub Vec<HeapStatisticRecord<'module_name>>);

pub fn heap_statistic(&self) -> HeapStatistic<'_>  {...}
```

### CODE REVIEW

File: tools/repl/src/repl.rs [hunk 1]
```diff
@@ -81,6 +81,7 @@ impl REPL {
             Some("e") | Some("envs") => self.show_envs(args),
             Some("f") | Some("fs") => self.show_fs(args),
             Some("i") | Some("interface") => self.show_interface(),
+            Some("h") | Some("heap") => self.show_heap_statistic(),
```
Comment (folex) [inline tools/repl/src/repl.rs:84]: h is usually for `help`. How about `s` for `stats`? And maybe also add `h` to show all possible commands?

File: tools/repl/src/repl.rs [hunk 2]
```diff
@@ -287,6 +294,7 @@ fn print_help() {
             u/unload <module_name>                                unload a Wasm module\n\
             c/call <module_name> <func_name> <args> [call_params] call function with given name from given module\n\
             i/interface                                           print public interface of all loaded modules\n\
+            h/heap                                                print heap sizes of all loaded modules\n\
```
Comment (folex) [inline tools/repl/src/repl.rs:297]: ditto

File: runtime/src/heap_statistic.rs [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+ */
+
+use std::fmt;
+use std::ops::Deref;
+
+/// Contains module name and a size of its heap in bytes.
+#[derive(Debug)]
+pub struct HeapStatisticRecord<'module_name> {
+    pub module_name: &'module_name str,
+    pub memory_size: usize,
```
Comment (folex) [inline runtime/src/heap_statistic.rs:24]: is it memory or heap? Would be nice to have consistent naming across `TomlFaaSModuleConfig` and `HeapStatisticRecord`

File: runtime/src/heap_statistic.rs [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use std::fmt;
+use std::ops::Deref;
+
+/// Contains module name and a size of its heap in bytes.
+#[derive(Debug)]
+pub struct HeapStatisticRecord<'module_name> {
```
Comment (folex) [inline runtime/src/heap_statistic.rs:22]: How about `ModuleHeapStat`? e.g.
```rust
struct ModuleHeapStat {
  name: str
  heap_size: str
}
```
This way, it's clear that this is about a single module

File: runtime/src/engine.rs [hunk 5]
```diff
@@ -136,6 +136,17 @@ impl Marine {
             .and_then(|module| module.export_record_type_by_id(record_id))
     }
 
+    /// Returns a heap size that all modules consume in bytes.
+    pub fn heap_statistic(&self) -> HeapStatistic<'_> {
```
Comment (folex) [inline runtime/src/engine.rs:140]: Is it an expensive operation or a cheap one? 

IMO it's a very important question for API, can you please mention that in the comment.

File: fluence-faas/src/faas.rs [hunk 6]
```diff
@@ -184,6 +185,11 @@ impl FluenceFaaS {
         FaaSInterface { modules }
     }
 
+    /// Return statistic of Wasm modules heap footprint.
+    pub fn heap_statistic(&self) -> HeapStatistic<'_> {
```
Comment (folex) [inline fluence-faas/src/faas.rs:189]: It's not a single value, so may be reflect that in the name? `heap_stats` would be nice. Or even `module_heap_stats`

File: runtime/src/engine.rs (hunk 5 above)
Comment (mikevoronov) [reply runtime/src/engine.rs:140]: It's a very cheap operation, I'll add a comment

**Pull request merged**